### PR TITLE
sl: Add recipe

### PR DESCRIPTION
**Pull request by tommyjanna**

### CODE REVIEW

**Comment by Begasus** [comment]
Upon checking Gentoo it looks correct :)
Did you try a build on x86_gcc2?

**Comment by tommyjanna** [comment]
Still new to haikuporter, was wondering how this is possible?

When I change my target architecture in `haikuports.conf` to x86_gcc2, and allow untested packages, haikuporter simply goes through some initial steps, then outputs "Nothing to do."

I believe I've covered the whole wiki, but can't figure this out... could you point me in the right direction? Do I need another install of Haiku?

**Comment by humdingerb** [comment]
You do try on an actual 32bit Haiku, right? Just changing the target arch in the haikuports.conf won't do... :)

File: app-misc/sl/sl-5.02.recipe [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+at all."
+HOMEPAGE="https://github.com/mtoyoda/sl"
+COPYRIGHT="1993,1998,2014 Toyoda Masashi"
+LICENSE="Toyoda"
+REVISION="1"
+SOURCE_URI="https://github.com/mtoyoda/$portName/archive/$portVersion/$portVersionedName.tar.gz"
+CHECKSUM_SHA256="1e5996757f879c81f202a18ad8e982195cf51c41727d3fea4af01fdcbbb5563a"
+SOURCE_DIR="sl-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
```
Comment (Begasus) [inline app-misc/sl/sl-5.02.recipe:13]: This is fine on x86_gcc2, no x86 primary architecture (any more) so:
```diff
- ARCHITECTURES="?x86_gcc2 x86 x86_64"
+ ARCHITECTURES="x86_gcc2 x86_64"
 ```

File: app-misc/sl/sl-5.02.recipe [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $binDir
+	cp sl $binDir
+
+	mkdir -p $dataDir/sl
+	cp LICENSE $dataDir/sl
```
Comment (Begasus) [inline app-misc/sl/sl-5.02.recipe:44]: No need for this LICENSE="Toyoda" takes care of putting the license in $dataDir/licenses :)
```diff
-
-	mkdir -p $dataDir/sl
-	cp LICENSE $dataDir/sl
```

**Comment by Begasus** [comment]
> When I change my target architecture in `haikuports.conf` to x86_gcc2, and allow untested packages, haikuporter simply goes through some initial steps, then outputs "Nothing to do."

My guess is that you are on x86_64 (64-bit Haiku), there is no target for x86_gcc2 there (yet), tested it and works ok, nice train btw :) Left some comments on a few changes, after those are fixed this can be merged.

**Comment by tommyjanna** [comment]
> You do try on an actual 32bit Haiku, right? Just changing the target arch in the haikuports.conf won't do... :)

Right, yeah. Was not sure if I should be expecting some sort of multilib support, so I guess I will need a 32-bit Haiku install. :)

> My guess is that you are on x86_64 (64-bit Haiku), there is no target for x86_gcc2 there (yet), tested it and works ok, nice train btw :) Left some comments on a few changes, after those are fixed this can be merged.

Edits have been made and pushed.

Thanks for taking a look through this!
Very kind community, I can't wait to do more work with you guys!

**Comment by Begasus** [comment]
> Very kind community

We try to be ;) Thanks for your contribution!

**Pull request merged**